test/unit/app/utils/test_user.py

```diff
@@ -53,3 +59,40 @@ def test_save_device_info(save_device_info, new_sempo_admin_user, serial_number,
     assert device is not None
     assert device.user_id is new_sempo_admin_user.id
 
+
+fake = Faker()
+fake.add_provider(phone_number)
+phone = partial(fake.msisdn)
+unactivated_user = partial(UserFactory, is_activated=False)
+standard_user = partial(UserFactory, pin='0000')
+
+
+@pytest.mark.parametrize("user_factory, user_input, expected_new_pin, expected_activation_status",
+                         [(unactivated_user, "0000", "0000", True)])
+def test_change_initial_pin(mocker, test_client, init_database, user_factory, user_input, expected_new_pin,
+                            expected_activation_status):
+    ua_user = user_factory()
+    ua_user.phone = phone()
+    assert ua_user.pin is None
+    assert ua_user.is_activated is False
+
+    user.send_sms = mocker.MagicMock()
+    change_initial_pin(ua_user, user_input)
+
+    assert ua_user.verify_pin(expected_new_pin) is True
```

Comment:
we don't need this test if the state machine test does essentially the same thing